### ISSUE TITLE
filter_shapenet files

### ISSUE THREAD

**Issue by AllanYangZhou** [liruiw/GA-DDPG#12]
Hi Lirui,

Thanks again for this awesome work. Is there any chance that the full shapenet training data can be made available? I assume these are the objects listed in `experiments/object_index/filter_shapenet.json`.

I tried following the README instructions for processing the shapes data myself but have run into various issues when trying to use OMG-Planner's `python -m real_world.process_shape`. For example, I'm having a little difficulty getting `meshlabserver` installed and running properly on my machine.

**Comment by liruiw** [comment]
Hello, unfortunately I think Shapenet objects require licenses so we cannot publicly share. Regarding data processing (it has been a while), I recalled it only required a python interface to the meshlab.

**Comment by AllanYangZhou** [comment]
Hi Lirui, thanks for the fast reply. I got meshlab working, though there are a few other issues I'm stuck on:

(1) I followed the [acronym](https://github.com/NVlabs/acronym#using-the-full-acronym-dataset) instructions to generate simplified meshes. I put each simplified mesh in the following format:
```
data
|----objects
|    |----Stapler_8d152be34b41785677937146265c551a
|    |    |    model_normalized.obj
|    |----[next object]
|    |    |    model_normalized.obj
```
After running `python -m OMG.real_world.process_shape -a -f Stapler_8d152be34b41785677937146265c551a` the size looks off. Maybe I missed an actual normalization step?
![example](https://user-images.githubusercontent.com/8732147/185338135-b2fcb158-6545-465e-a7bb-e21a1c7fef29.png)

(2) Additionally, where do I put the acronym grasps? Acronym gives a folder of h5 files such as `Xbox360_eabea01e420a96a3c55b18f95951c1e1_0.012159769145183007.h5`. Do I put those in `data/grasps/simulated/`? The existing grasps there are in `.npy` format.

**Comment by liruiw** [comment]
I think the scale should come from the acronym dataset "/object/scale" (and there should be a resizing step here). Could you try visualizing the grasps in acronym dataset with this [code](https://github.com/NVlabs/6dof-graspnet/blob/master/grasp_data_reader.py) and this [repo](https://github.com/jsll/pytorch_6dof-graspnet), and see if it looks good to you? Yes, the grasps should be in /simulated.

**Comment by liruiw** [comment]
I added some scripts from my old codebase to the repo. Feel free to take a look at those and see if it is helpful.

**Comment by AllanYangZhou** [comment]
Manually resizing the OBJ files using `trimesh.apply_scale` before `process_shape` (using the value `/object/scale` from acronym dataset) appears to work for most objects (though some of them like `TV_dadf3f55c02c4dbe71357e42f9203ca5` don't seem to appear at all when I load them into the pybullet environment).

For the grasps, I can't seem to get expert planning to work (0% SR). It seems like expert planning uses [this code](https://github.com/liruiw/OMG-Planner/blob/8425a0d77cbbd568cc078e6e9458a7250a4fb110/omg/planner.py#L457) which is for YCB objects. Does it work the same for acronym objects?

I'll try the visualization scripts you linked when I have some time in the next few days, thanks!

**Comment by AllanYangZhou** [comment]
Some examples:
![5](https://user-images.githubusercontent.com/8732147/185564452-8cc0b503-25cc-48f9-8db8-e2bebc2591f9.gif)
![8](https://user-images.githubusercontent.com/8732147/185564529-416f562e-1a49-4035-9845-199d4eb5e612.gif)
![3](https://user-images.githubusercontent.com/8732147/185564567-f1fbce49-23a2-484a-bbd4-67f66c09db83.gif)
![2](https://user-images.githubusercontent.com/8732147/185564771-fe50c1e1-a1f8-4a66-8881-67a3a6ad16ce.gif)
(In the last one the object doesn't seem to appear at all)

**Comment by liruiw** [comment]
Yes the grasp loading code should also work for shapenet objects. It looks like the object is not given a stable pose before simulation. Probably would require some debugging in the loading object and env reseting code even without the robot or planner

**Comment by AllanYangZhou** [comment]
That's helpful, I did some more debugging at reset time. It looks like it's still some sort of geometry or processing issue though. The object is dropped in a fairly stable pose, but when it hits the table the physics don't look right at all and it's center of mass seems off.
![Laptop_dd67a2225e7b61c162682d6b7b4ec83](https://user-images.githubusercontent.com/8732147/185735813-38dfe3de-297b-4c2e-b550-727036fa7899.gif)

**Comment by liruiw** [comment]
Right it seems that some of the physics parameters might be off. Maybe see if bullet provides some ways to recompute center of mass and inertia of objects after loading.
